doc/interoperability/sjs-defined-js-classes.md

```diff
@@ -1,26 +1,44 @@
 ---
 layout: doc
-title: Scala.js-defined JS classes
+title: Non-native JS types (aka Scala.js-defined JS types)
 ---
 
-As explained in detail in the [guide to write facade types](./facade-types.html), classes, traits and objects inheriting from `js.Any` are native by default.
-To implement a JavaScript class in Scala.js, it should be annotated with `@ScalaJSDefined`:
+A non-native JS type, aka Scala.js-defined JS type, is a JavaScript type implemented in Scala.js code.
+This is in contrast to native JS types, described in [the facade types reference](./facade-types.html), which represent APIs implemented in JavaScript code.
+
+## About `@ScalaJSDefined`
+
+In Scala.js 0.6.x, the `@ScalaJSDefined` is necessary to declare a non-native JS type, also called a Scala.js-defined JS type.
+Starting from Scala.js 1.x however, the annotation is not necessary anymore.
+Since Scala.js 0.6.17, you can opt-in for the new semantics of 1.x where `@ScalaJSDefined` is not necessary, by giving the option `-P:scalajs:sjsDefinedByDefault` to scalac.
+In an sbt build, this is done with
+
+{% highlight scala %}
+scalacOptions += "-P:scalajs-sjsDefinedByDefault"
+{% endhighlight %}
+
+The present documentation assumes that you are using this option (or Scala.js 1.x).
+Code snippets mention the necessary `@ScalaJSDefined` in comments as a reference for older versions.
+
+## Defining a non-native JS type
+
+Any class, trait or object that extends `js.Any` is a JS type:
 
 {% highlight scala %}
 import scala.scalajs.js
-import js.annotation._
+import scala.scalajs.js.annotation._
 
-@ScalaJSDefined
+// @ScalaJSDefined
 class Foo extends js.Object {
```

Comment:
You probably want to mention somehow that `js.Object` extends `js.Any`. Otherwise the line above (.. that extends `js.Any` is a ...) looks like a typo. We don't want that :)